plugins/module_utils/utils.py

```diff
@@ -913,6 +913,10 @@ def _update_object(self, data):
         serialized_nb_obj = self.nb_object.serialize()
         updated_obj = serialized_nb_obj.copy()
         updated_obj.update(data)
+        if "custom_fields" in serialized_nb_obj:
+            custom_fields = serialized_nb_obj.get("custom_fields", {})
+            shared_keys = custom_fields.keys() & data.get("custom_fields", {}).keys()
+            serialized_nb_obj["custom_fields"] = {key: custom_fields[key] for key in shared_keys if custom_fields[key] is not None}
```

Comment:
Example of why this is failing tests:

After `serialized_nb_obj = self.nb_object.serialize()`:
```python
serialized_nb_obj = {
  "name": "foo",
  "custom_fields": {
    "bar": None
  }
}
```

After `updated_obj = serialized_nb_obj.copy()`:
```python
updated_obj = {
  "name": "foo",
  "custom_fields": {
    "bar": None
  }
}
```

After `serialized_nb_obj["custom_fields"] = {key: custom_fields[key] for key in shared_keys if custom_fields[key] is not None}`:
```python
serialized_nb_obj = {
  "name": "foo",
  "custom_fields": {}
}
```

Hence, `serialized_nb_obj` is not equal to `updated_obj`. You need to null out the custom fields in `updated_obj` too.